exoctk/atmospheric_retrievals/platon_wrapper.py

```diff
@@ -0,0 +1,268 @@
+    required_parameters = ['Rs', 'Mp', 'Rp', 'T']
+    param_types = [float, float, float, int]
+    for param, param_type in zip(required_parameters, param_types):
+        assert param in params, '{} missing from parameter file'.format(param)
+        assert type(params[param]) == param_type, '{} is not of type {}'.format(param, param_type)
+
+    # Make sure the optional keywords are of proper type
+    optional_parameters = ['logZ', 'CO_ratio', 'log_cloudtop_P', 'log_scatt_factor',
+                           'scatt_slope', 'error_multiple', 'T_star']
+    param_types = [int, float, int, int, int, int, int]
+    for param, param_type in zip(optional_parameters, param_types):
+        if param in params:
+            assert type(params[param]) == param_type, '{} is not of type {}'.format(param, param_type)
+
+
+class PlatonWrapper():
+    """Class object for running the platon atmospheric retrieval
+    software."""
+
+    def __init__(self):
+        """Initialize the class object."""
+
+        self.retriever = Retriever()
+        self.output_results = 'results.dat'
+        self.output_plot = 'corner.png'
+
+    def make_plot(self):
+        """Create a corner plot that shows the results of the retrieval."""
+
+        if self.method == 'emcee':
+            fig = corner.corner(self.result.flatchain, range=[0.99] * self.result.flatchain.shape[1],
+                        labels=self.fit_info.fit_param_names)
+
+        elif self.method == 'multinest':
+            fig = corner.corner(self.result.samples, weights=self.result.weights,
+                                range=[0.99] * self.result.samples.shape[1],
+                                labels=self.fit_info.fit_param_names)
+
+        # Save the results
+        self.output_plot = '{}_corner.png'.format(self.method)
+        fig.savefig(self.output_plot)
+        print('Corner plot saved to {}'.format(self.output_plot))
+
+    def retrieve_emcee(self):
+        """Perform the atmopsheric retrieval via emcee."""
+
+        self.method = 'emcee'
+        self.result = self.retriever.run_emcee(self.bins, self.depths, self.errors, self.fit_info)
+
+    def retrieve_multinest(self):
+        """Perform the atmopsheric retrieval via multinested sampling."""
+
+        self.method = 'multinest'
+        self.result = self.retriever.run_multinest(self.bins, self.depths, self.errors, self.fit_info, plot_best=False)
+
+    def save_results(self):
+        """Save the results of the retrieval to an output file."""
+
+        # Save the results
+        self.output_results = '{}_results.dat'.format(self.method)
+        with open(self.output_results, 'w') as f:
+            f.write(str(self.result))
+        print('Results file saved to {}'.format(self.output_results))
+
+    def set_parameters(self, params):
+        """Set necessary parameters to perform the retrieval.
+
+        Required parameters include ``Rs``, ``Mp``, ``Rp``, and ``T``.
```

Comment:
@bourque Neat. I was gonna ask that you enumerate the required and optional parameters in the most logical one of these functions.